### PR TITLE
Writting e2e test for the Node Observability Run

### PR DESCRIPTION
**Pull request by ayesha54**

### CODE REVIEW

File: test/e2e/nodeobservabiltiyrun_test.go [hunk 1]
```diff
@@ -35,6 +43,92 @@ import (
...
+	}
+	return &nodeObs
+}
+func TestIfProfilingIsFinished(t *testing.T) {
+	nodeObsRun := ensureNodeObservabilityRunResource()
+	err := kubeClient.Get(context.TODO(), types.NamespacedName{Name: nodeobservability, Namespace: operandNamespace}, nodeObsRun)
+	if err != nil {
+		t.Fatalf("Failed to get NodeObservabilityRun Resource in ns %s: %v\n", operandNamespace, err)
+	}
+	time.Sleep(30 * time.Second)
```
Comment (josefkarasek) [inline test/e2e/nodeobservabiltiyrun_test.go:112]: It's not guaranteed the scraping will finish in exactly 30 seconds.

We should poll instead:
1) create NodeObsRun
2) poll NodeObsRun and check .Status.FinishedTimestamp
3) success when finished under timeout seconds; len(.Status.FailedAgents) = 0

File: test/e2e/nodeobservabiltiyrun_test.go [hunk 2]
```diff
@@ -35,6 +43,92 @@ import (
...
+	}
+}
+func TestIfAnyFailedAgents(t *testing.T) {
+	nodeObsRun := ensureNodeObservabilityRunResource()
+	err := kubeClient.Get(context.TODO(), types.NamespacedName{Name: nodeobservability, Namespace: operandNamespace}, nodeObsRun)
+	if err != nil {
+		t.Fatalf("Failed to get NodeObservabilityRun Resource in ns %s: %v\n", operandNamespace, err)
+	}
+	failedAgents := nodeObsRun.Status.FailedAgents
+	if failedAgents != nil {
```
Comment (josefkarasek) [inline test/e2e/nodeobservabiltiyrun_test.go:127]: `failedAgents != nil && len(failedAgents) > 0

File: test/e2e/nodeobservabiltiyrun_test.go [hunk 3]
```diff
@@ -35,6 +44,101 @@ import (
...
+			Name:      nodeobservability,
+			Namespace: operandNamespace,
+		},
+		Spec: spec,
+	}
+	return &nodeObs
+}
+func TestIfNodeObservabilityRunFinished(t *testing.T) {
+	if err := IfFinished(t); err != nil {
+		t.Errorf("Node Observability Run is Failed: %v", err)
```
Comment (lmzuccarelli) [inline test/e2e/nodeobservabiltiyrun_test.go:109]: nit: Change Node Observability Run is Failed -> Node Observability Run has Failed

File: test/e2e/nodeobservabiltiyrun_test.go [hunk 4]
```diff
@@ -35,6 +44,101 @@ import (
...
+	return wait.PollImmediate(1*time.Second, 1*time.Minute, func() (bool, error) {
+		nodeObsRun := ensureNodeObservabilityRunResource()
+		err := kubeClient.Get(context.TODO(), types.NamespacedName{Name: nodeobservability, Namespace: operandNamespace}, nodeObsRun)
+		if err != nil {
+			t.Logf("Failed to get NodeObservabilityRun Resource in ns %s: %v\n", operandNamespace, err)
+			return false, nil
+		}
+		isFinished := nodeObsRun.Status.FinishedTimestamp
+		if isFinished != nil && !isFinished.IsZero() && len(nodeObsRun.Status.FailedAgents) == 0 {
+			fmt.Printf("Node Observability Run is Finished Successfully")
```
Comment (lmzuccarelli) [inline test/e2e/nodeobservabiltiyrun_test.go:123]: nit: Change Node Observability Run is Finished Successfully -> Node Observability Run has Finished Successfully

**Comment by lmzuccarelli** [comment]
@ayesha54 - could you also fix this `test/e2e/nodeobservabiltiyrun_test.go:137:5  gosimple  S1009: should omit nil check; len() for nil slices is defined as zero`

**Comment by lmzuccarelli** [comment]
@ayesha54 - only noticed now, Could you change the file name `test/e2e/nodeobservabiltiyrun_test.go` to `test/e2e/nodeobservabilityrun_test.go` - just for conciseness

**Comment by lmzuccarelli** [comment]
This PR has been replaced by https://github.com/openshift/node-observability-operator/pull/36 - so I'm closing it

**Pull request closed without merging**